src/kit.ts

```diff
@@ -580,6 +595,70 @@ async function collectDevBatVars(devbat: string, args: string[], major_version:
   return vars;
 }
 
+/**
+ * Gets the environment variables set by a shell script.
+ * @param kit The kit to get the environment variables for
+ */
+export async function getShellScriptEnvironment(kit: Kit): Promise<Map<string, string>|undefined> {
+  console.assert(kit.environmentVariablesShellScript);
+  const fname = Math.random().toString() + (process.platform == 'win32' ? '.bat' : '.sh');
```

Comment:
We prefer it if you don't abbreviate variable names.  I think you mean `filename` here.

Reply:
Okay